### PR TITLE
Compilation changes

### PR DESCRIPTION
**Pull request by cmeiklejohn**
Remove riak_search dependency, and trade up for riak_kv.
Address warnings in tests.
Address a ton of typos.

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -6,6 +6,6 @@
 {deps, [
         {lager, "2.0.*", {git, "git://github.com/basho/lager.git", {tag, "2.0.1"}}},
         {ebloom, ".*", {git, "git://github.com/basho/ebloom.git", {branch, "develop"}}},
-        {riak_search, ".*", {git, "git://github.com/basho/riak_search.git", {branch, "develop"}}},
+        {riak_kv, ".*", {git, "git://github.com/basho/riak_kv.git", {branch, "develop"}}},
```
Comment (bookshelfdave) [inline rebar.config:9]: @jaredmorrow ^, cool with you?

Comment (jaredmorrow) [inline rebar.config:9]: I didn't put the search dep in there to begin with.  At some point you must have needed search?  If you don't need it, yeah remove it.

Comment (bookshelfdave) [inline rebar.config:9]: I only asked because you went caremad on deps last month.

Comment (cmeiklejohn) [inline rebar.config:9]: Yes, remove the dep!

Comment (jaredmorrow) [inline rebar.config:9]: You can do whatever you want, just don't **add** anymore deps!

**Comment by bookshelfdave** [comment]
+1

**Pull request merged**